packages/components/src/index.js

```diff
@@ -98,7 +98,7 @@ export {
 	Slot,
 	Fill,
 	Provider as SlotFillProvider,
-	Consumer as __experimentalSlotFillConsumer,
+	useSlot as __experimentalUseSlot,
```

Comment:
If I got it right, we only had to expose `Consumer` for the cases where we needed to check if slots or fills have been registered.

On `Popover`, we check the presence of the slot in order to render the contents within a `<Fill>`:
https://github.com/WordPress/gutenberg/blob/59b07cc57adff90630fc9d5cf2317269a0f4f158/packages/components/src/popover/index.js#L606-L622

On `BlockInspector`, we check the presence of fills so we can conditionally render the slot wrapper:
https://github.com/WordPress/gutenberg/blob/59b07cc57adff90630fc9d5cf2317269a0f4f158/packages/block-editor/src/components/block-inspector/index.js#L72-L86

Reply:
Yes, I like the hook better, nice work here.